### ISSUE TITLE
Files used in notebooks are not all in the Dataverse yet

### ISSUE THREAD

**Issue by gaelforget** [AIRCentre/JuliaEO#36]

**Comment by gaelforget** [comment]
Regarding data uploads.

See https://github.com/gdcc/Dataverse.jl/issues/13 , https://github.com/IQSS/dataverse/issues/9298

**Comment by visr** [comment]
> We'd like to document how much gets downloaded, for users who may have limited internet access may want to know.

In [juliageo.ipynb](https://github.com/AIRCentre/JuliaEO/blob/1ba0755dc027e73e6d1b3a863259b5b91faee798/notebooks/plenary_sessions/The_Power_of_JuliaGeo/juliageo.ipynb) all data gets downloaded, which is about 100MB gets downloaded. This is almost entirely the geometries from GADM, which it only needs to download once. A Manifest.toml is also uploaded for reproducability.

**Comment by gaelforget** [comment]
Btw, I anticipate a nice of set of file types will end up in the JuliaEO dataverse repo.  

It likely would be useful to have a few examples out of zip. To demo previewing and maybe lazy access functionalities, for example.

- already there : netcdf (arrays), geotiff (rasters), and geojson (polygons)
- possible / likely : csv, JLD2 (Julia / HDF5), zarr (arrays, cloud optimized), parquet (geospatial, cloud optimized), geotiff & nectdf (more, cl;oud optimized), Dockerfile (text but automated image build would be nice), data cubes, ...

Looping in @pdurbin @atrisovic @felixcremer @rafaqz @Alexander-Barth just in case

**Comment by gaelforget** [comment]
quoting from @pdurbin on different platform, about the auto-unzip functionality for upload :

_you have to double zip. This is how I do it for [my dataset](https://github.com/pdurbin/open-source-at-harvard-primary-data):
```
zip -r primary-data.zip primary-data -x '**/.*' -x '**/__MACOSX'
zip -r outer.zip primary-data.zip -x '**/.*' -x '**/__MACOSX'
```
_

**Comment by gaelforget** [comment]
For supporting notebooks that require data from `Dataverse`, the simplest thing I could think of would be to have one `tar.gz` or `zip` file associated with the notebook folder in GitHub.

e.g. , `Data_Visualizations_with_Makie.tar.gz` or some `zip` version

I think this makes life easy on the self-guided user. 

Are there strong preferences between `tar.gz` and `zip`?

Supposedly we can add software on the Dataverse or Julia side to extract meta data from `tar.gz` and preview in UI / without downloading the file. 

Would be good also to grab the Dockerfile as preview of a Docker image like [DockerImage.tar.gz](https://dataverse.harvard.edu/file.xhtml?fileId=6880857&version=5.0) I think.

**Comment by pdurbin** [comment]
> ## sessions that used custom Docker images
> 
> We'd like to collect these too and upload them to the Dataverse as well if possible. The central one that's meant to support running all notebooks (ideally) has already been posted there.

@gaelforget @visr last week we enabled a Binder button on Harvard Dataverse:

- https://github.com/IQSS/dataverse.harvard.edu/issues/208

It looks like this on the "Global Workshop on Earth Observation with Julia 2023" dataset at https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/OYBLGK

![Screen Shot 2023-02-01 at 3 34 02 PM](https://user-images.githubusercontent.com/21006/216157643-29e92c62-2ee6-4826-8d73-2952d35d064d.png)

I'm scared to click it, though, because the Docker image is 1.7 GB! 😅 Binder will try to download all the files in the dataset, including that giant one.

Binder supports defining your own Dockerfile (an alternative to uploading the image itself): https://mybinder.readthedocs.io/en/latest/tutorials/dockerfile.html

Is this something you'd like to try?

For more on Binder from the Dataverse perspective: https://guides.dataverse.org/en/5.12.1/admin/integrations.html#binder

**Comment by gaelforget** [comment]
For `Land_Cover_Classification_of_Earth_Observation_images` the files used, depending on user choice, are :

- Small : RGB (133M, 3 bands)
 - Large : MS (2.8G, 13 bands)

**Comment by gaelforget** [comment]
For `RF_classification_using_marida` the files used are :

- 551M   `Bands_Indices-S2`
- 57M   `Train_Test-Datasets`

**Comment by EmanuelCastanho** [comment]
Yes, `Train_Test-Datasets` is a subset created by me from the original MARIDA dataset.
According to their MIT Licence and Creative Commons Attribution 4.0 International I think it is fine to include this subset in the Dataverse.

Please include this citation to the original dataset, if possible: Kikaki K, Kakogeorgiou I, Mikeli P, Raitsos DE, Karantzalos K (2022) MARIDA: A benchmark for Marine Debris detection from Sentinel-2 remote sensing data. PLoS ONE 17(1): e0262247. https://doi.org/10.1371/journal.pone.0262247

`Bands_Indices-S2` can be included in the Dataverse.

@gaelforget

**Comment by gaelforget** [comment]
> Yes, `Train_Test-Datasets` is a subset created by me from the original MARIDA dataset. According to their MIT Licence and Creative Commons Attribution 4.0 International I think it is fine to include this subset in the Dataverse.
> 
> Please include this citation to the original dataset, if possible: Kikaki K, Kakogeorgiou I, Mikeli P, Raitsos DE, Karantzalos K (2022) MARIDA: A benchmark for Marine Debris detection from Sentinel-2 remote sensing data. PLoS ONE 17(1): e0262247. https://doi.org/10.1371/journal.pone.0262247
> 
> `Bands_Indices-S2` can be included in the Dataverse.
> 
> @gaelforget

Done. Thanks!

Dataverse : https://doi.org/10.7910/DVN/OYBLGK
Zenodo : https://doi.org/10.5281/zenodo.8113073

**Comment by gaelforget** [comment]
automated data downloads have now been implemented for some notebooks

#43 #44 #45
